File: qc-core/src/main/java/com/eager/questioncloud/core/domain/creator/model/Creator.java
package com.eager.questioncloud.core.domain.creator.model;

import com.eager.questioncloud.core.domain.creator.vo.CreatorProfile;
import com.eager.questioncloud.core.domain.user.model.User;
import lombok.Builder;
import lombok.Getter;

@Getter
public class Creator {
    private Long id;
    private User user;
    private CreatorProfile creatorProfile;

    @Builder
    public Creator(Long id, User user, CreatorProfile creatorProfile) {
        this.id = id;
        this.user = user;
        this.creatorProfile = creatorProfile;
    }

    public static Creator create(User user, CreatorProfile creatorProfile) {
        return Creator.builder()
            .user(user)
            .creatorProfile(creatorProfile)
            .build();
    }

    public void updateProfile(CreatorProfile creatorProfile) {
        this.creatorProfile = creatorProfile;
    }
}

File: qc-core/src/main/java/com/eager/questioncloud/core/domain/creator/model/Creator.kt
package com.eager.questioncloud.core.domain.creator.model

import com.eager.questioncloud.core.domain.question.enums.Subject


class Creator(
    val id: Long = 0,
    val userId: Long,
    var mainSubject: Subject,
    var introduction: String,
) {
    fun updateProfile(mainSubject: Subject, introduction: String) {
        this.mainSubject = mainSubject
        this.introduction = introduction
    }

    companion object {
        fun create(userId: Long, mainSubject: Subject, introduction: String): Creator {
            return Creator(userId = userId, mainSubject = mainSubject, introduction = introduction)
        }
    }
}
